API/Controllers/RoleController.cs

```diff
@@ -235,7 +236,19 @@ public async Task<IActionResult> DeleteRole(int roleId)
                 };
                 return NotFound(problem);
             }
-            await roleService.RemoveAsync(roleId);
+
+            if(userService.UserWithRoleExists(role))
+            {
+                ProblemDetails problem = new ProblemDetails
+                 {
+                     Title = "Role is still assigned.",
+                     Detail = "The role is still assigned to a user.",
```

Comment:
Weird error message don't you think? Does this error message tell the user what went wrong?